### DISCUSSION TITLE
Encryption verification is failed

### DISCUSSION THREAD

**Discussion by HI448267** [ONDC-Official/developer-docs#45] [Q&A]
Hi @92shreyansh  , @subodhparamanik

Getting below issue when we hit the subscribe, can please help me to solve the issue.

URL : https://staging.registry.ondc.org/subscribe
"https://mtpre.adityabirlahealth.com/healthinsurance/ondc/on_subscribe : Encryption verification is failed"
https://mtpre.adityabirlahealth.com/ondc-site-verification.html

Below steps are followed :
1. Keys generated : 
Signing_private_key: Z+Ela93Y0kLk5bO0w75NCehV3FzxTHViVVTW46BbNalxebH2GS8IQxAk8qoIHH71GkmvpNRqAOoHOQ2V3ixUOg==
Signing_public_key:  cXmx9hkvCEMQJPKqCBx+9RpJr6TUagDqBzkNld4sVDo=
Crypto_Privatekey:   MC4CAQAwBQYDK2VuBCIEIKg8RhOkdulgBhQ/hTa5q2XI0dF9GizCcN/p8mIp4PJ3
Crypto_Publickey:    MCowBQYDK2VuAyEA8xRoLUABA4QjZMUNnYeaLiIgQkwX+JT+8Fq1zrtXB2k=

2. Changed in index.js and ondc-site-verification.html is created with below details :
const PRIVATE_KEY_1 =
  'MC4CAQAwBQYDK2VuBCIEIKg8RhOkdulgBhQ/hTa5q2XI0dF9GizCcN/p8mIp4PJ3';
const PUBLIC_KEY_1 =
  'MCowBQYDK2VuAyEA8xRoLUABA4QjZMUNnYeaLiIgQkwX+JT+8Fq1zrtXB2k=';
const REQUEST_ID = '31baa06df10a023c85e5cc621b787f05';
const SIGNING_PRIVATE_KEY =
  'Z+Ela93Y0kLk5bO0w75NCehV3FzxTHViVVTW46BbNalxebH2GS8IQxAk8qoIHH71GkmvpNRqAOoHOQ2V3ixUOg==';

3. SIGNED_UNIQUE_REQ_ID is generted :
"eGPdXjl/a0TJwcMe6yghZtfpm3HG6VLtMTsCeBof2RFn8Eq/WCeYhSEjSIIcqMS6fLFWZmP/0N2x6TvXj8slBw=="

4. Below is the Request :
{
  "context": {
    "operation": {
      "ops_no": 2
    }
  },
  "message": {
    "request_id": "31baa06df10a023c85e5cc621b787f05",
    "timestamp": "2023-11-08T08:01:54.101Z",
    "entity": {
      "gst": {
        "legal_entity_name": "ADITYA BIRLA HEALTH INSURANCE CO LIMITED",
        "business_address": "9th Floor, Unit no 903-A, One Indiabulls Center ,Tower-1, Jupiter Mill Compound, S.B.Marg, Elphinstone Road, Mumbai City, Maharashtra, 400013",
        "city_code": [
          "*"
        ],
        "gst_no": "2XXXXXXXXXXXXXN"
      },
      "pan": {
        "name_as_per_pan": "ADITYA BIRLA HEALTH INSURANCE CO LIMITED",
        "pan_no": "AXXXXXXXXG",
        "date_of_incorporation": "01/07/2017"
      },
      "name_of_authorised_signatory": "Dhruv Shankar",
      "address_of_authorised_signatory": "11th Floor, Sai Nagar, Anand Nagar, Kasarwadawali, Ghodbunder Road, Thane (West), Mumbai, 400615",
      "email_id": "dhruv.shankar@adityabirlacapital.com",
      "mobile_no": 9XXXXXXXX9,
      "country": "IND",
      "subscriber_id": "mtpre.adityabirlahealth.com",
      "unique_key_id": "31baa06df10a023c85e5cc621b787f05",
      "callback_url": "/healthinsurance/ondc",
      "key_pair": {
        "signing_public_key": "cXmx9hkvCEMQJPKqCBx+9RpJr6TUagDqBzkNld4sVDo=",
        "encryption_public_key": "MCowBQYDK2VuAyEAduMuZgmtpjdCuxv+Nc49K0cB6tL/Dj3HZetvVN7ZekM=",
        "valid_from": "2023-10-30T13:44:54.101Z",
        "valid_until": "2024-10-31T20:44:54.101Z"
      }
    },
    "network_participant": [
      {
        "subscriber_url": "/healthinsurance/ondc",
        "domain": "ONDC:FIS13",
        "type": "sellerApp",
        "msn": false,
        "city_code": [
          "*"
        ]
      }
    ]
  }
}

5. Getting below isse :
{
    "message": {
        "ack": {
            "status": "NACK"
        }
    },
    "error": {
        "type": "CORE-ERROR",
        "code": "131",
        "path": null,
        "message": "https://mtpre.adityabirlahealth.com/healthinsurance/ondc/on_subscribe : Encryption verification is failed"
    }
}

**Comment by subodhParamanik** [comment]
@HI448267 
Please used Crypto_Publickey: MCowBQYDK2VuAyEA8xRoLUABA4QjZMUNnYeaLiIgQkwX+JT+8Fq1zrtXB2k= as your encryption_public_key
in subscribe api.

**Reply by HI448267** [reply]
Thanks subodh, 
getting same issue, refer attachment.
![Screenshot 2023-11-08 130820](https://github.com/ONDC-Official/developer-docs/assets/139313705/b90b0c8f-371c-4436-b93d-ec4a7ae2d9ef)

**Comment by HI448267** [comment]
Thanks subodh,
After changing getting same issue, refer attachment.
![Screenshot 2023-11-08 130820](https://github.com/ONDC-Official/developer-docs/assets/139313705/6f9d51e0-e57a-4314-8b49-0c9abdcdbb52)

**Reply by subodhParamanik** [reply]
@HI448267 
Run Node Index.js folder in your local server and paste the generated [ondc-site-verification.html] code to your server page (https://mtpre.adityabirlahealth.com/ondc-site-verification.html) and recall the subscribe api. Here you get SIGNED_UNIQUE_REQ_ID 
Example:-
http://localhost:3000/ondc-site-verification.html

**Reply by HI448267** [reply]
@subodhParamanik  can you please help me.

**Reply by 92shreyansh** [reply]
Also check if you are receiving the on_subscribe call with required challange and while decrypting use your own private encryption key and Registry's public encryption key to decrypt the logic as in the utility [here](https://github.com/ONDC-Official/reference-implementations/tree/main/utilities/on_subscibe-service/node)

**Comment by HI448267** [comment]
Hi @subodhParamanik ,
Followed the given steps still getting same error, attached screenshot for reference.
![1](https://github.com/ONDC-Official/developer-docs/assets/139313705/7c0899ce-ef3a-4d90-bfbd-5565f65033e7)
![2](https://github.com/ONDC-Official/developer-docs/assets/139313705/3857b410-1d76-4a64-bb35-3c461997f1d3)
![3](https://github.com/ONDC-Official/developer-docs/assets/139313705/2cb54f3b-4637-4048-9cf7-6cf21589a89d)

Is it possible to connect on Google Meet or Teams.?

**Comment by 92shreyansh** [comment]
Request to move this disscussion to the dedicated thread of subscribe https://github.com/ONDC-Official/developer-docs/discussions/31

**Comment by rasika-sarang** [comment]
@HI448267 Is your issue resolved ? I am also facing the same issue. Please let me know the resolution if you have the fix. Thank You

**Reply by ramsabi** [reply]
Hi rasika, I am having the same issue. Were you able tos olve. If yes, will be grateful if you could help.
